AG-14059 - Add console log component

taktran opened the pull request:
## Changes

* Refactor darkmode snippet insertion, so it's a generic function that gets passed into `frameworkFilesGenerator`
* Add console log override for examples
  * Add snippet in example generator, and remove it from view in `CodeViewer.tsx`
  * Send console message to parent window
  * Listen to iframe console messages and display them underneath the example runner
* Replace all `window.alert`s in examples to use `console.log`

taktran committed:
AG-14059 - Refactor darkmode snippet

* Extract darkmode snippet code from frameworkFilesGenerator
* Refactor snippet regex

taktran committed:
AG-14059 - Add console log override for examples

* Add snippet in example generator, and remove it from view in `CodeViewer.tsx`
* Send console message to parent window
* Listen to iframe console messages and display them underneath the example runner

taktran committed:
AG-14059 - Replace all alerts with console logs

alantreadway commented on packages/ag-charts-website/src/content/docs/events/_examples/chart-click-event/main.ts:
```diff
@@ -34,10 +32,10 @@ const options: AgChartOptions = {
     listeners: {
         click: (_event: AgChartClickEvent) => {
-            window.alert('[click]');
+            console.log('[click] 216');
```
```suggestion
            console.log('[click]');
```

taktran replied:
Ah, caching hack 🤦🏼‍♂️ - good catch 😄

alantreadway reviewed (approved):
LGTM aside from one typo

taktran committed:
AG-14059 - Use pre styles for console log + adapt border radius

taktran committed:
AG-14059 - Get `hasExampleConsoleLog` at build time

So there is no layout shift on load if there is a console log component.

Also clean up console log component styles

taktran committed:
AG-14059 - Ignore angular messages

taktran committed:
AG-14059 - Hide no logs message once logs appear

taktran committed:
AG-14059 - Scroll to the bottom of the logs when new logs are added

taktran committed:
AG-14059 - Clean up example logger styles

github-actions[bot] committed:
Update E2E snapshots.

markdurrant committed:
AG-14059 - Update console darkmode background colour
